src/core/createEventManager.js

```diff
@@ -93,8 +95,38 @@ export default ({
             request,
             runOnResponseCallbacks: onResponseCallbackAggregator.call,
             runOnRequestFailureCallbacks:
-              onRequestFailureCallbackAggregator.call,
-            serverState
+              onRequestFailureCallbackAggregator.call
+          });
+        });
+    },
+    applyHandles(event, options = {}) {
+      const {
+        renderDecisions = false,
+        decisionScopes = [],
+        handles = []
+      } = options;
+
+      const payload = createDataCollectionRequestPayload();
+      const request = createDataCollectionRequest(payload);
+      const onResponseCallbackAggregator = createCallbackAggregator();
+      const onRequestFailureCallbackAggregator = createCallbackAggregator();
+
+      return lifecycle
+        .onBeforeEvent({
+          event,
+          renderDecisions,
+          decisionScopes,
+          onResponse: onResponseCallbackAggregator.add,
+          onRequestFailure: onRequestFailureCallbackAggregator.add
```

Comment:
There is no way the request could fail because it is not going to be made. This could be a no-op function